src/main/java/com/example/provider/VehicleServlet.java

```diff
@@ -82,35 +88,46 @@ public VehicleServlet(
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     ServletUtils.setStandardResponseHeaders(response);
-
+    
     String vehicleId;
 
-    try {
-      vehicleId = ServletUtils.getEntityIdFromRequestPath(request);
-    } catch (IllegalArgumentException e) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "No Vehicle id provided.");
-      return;
-    }
+    vehicleId = ServletUtils.getEntityIdFromRequestPath(request);
 
-    logger.info(String.format("Getting vehicle with vehicleID: %s", vehicleId));
+    if (vehicleId != null && !vehicleId.isEmpty()) {
```

Comment:
nit: is there a case where `vehicleId` would be null?

Reply:
Done